### PR TITLE
fix crashing bug

### PR DESCRIPTION
**Pull request by Salam-Dalloul**

### CODE REVIEW

**Review by ddelpiano — APPROVED** [review]

File: client/src/helpers/ImagesTreeViewHelpers.js [hunk 1]
```diff
@@ -16,20 +16,25 @@ export const DoDataPreprocessing = (filteredActivityMaps, currentExperimentName)
   Object.values(filteredActivityMaps).map(obj => {
     // if experiment key exists
     if (obj.experiment !== undefined) {
```
Comment (ddelpiano) [inline client/src/helpers/ImagesTreeViewHelpers.js:18]:
```
if (obj.experiment === undefined || obj.experiment === true) {...}
else {...}
```

wouldn't the above be better?

File: client/src/helpers/ImagesTreeViewHelpers.js [hunk 2]
```diff
@@ -16,20 +16,25 @@ export const DoDataPreprocessing = (filteredActivityMaps, currentExperimentName)
...
-      if (obj.experiment === true) {
+      if(obj.experiment === true){
         // experiment is true
-        obj.hierarchy = obj.hierarchy === undefined || obj.hierarchy.includes(currentExperimentName)
-          ? obj.hierarchy
-          : [currentExperimentName, ...obj.hierarchy];
-      } else {
+        if(obj.hierarchy === undefined){ // no hierarchy key
```
Comment (ddelpiano) [inline client/src/helpers/ImagesTreeViewHelpers.js:21]: This logic and the one above if experiment is false is the same, why the need of the if else then?

File: client/src/helpers/ImagesTreeViewHelpers.js [hunk 3]
```diff
@@ -16,20 +16,25 @@ export const DoDataPreprocessing = (filteredActivityMaps, currentExperimentName)
...
     } else {
       // experiment key not exists
-      obj.hierarchy = obj.hierarchy === undefined || obj.hierarchy.includes(currentExperimentName)
-        ? obj.hierarchy
-        : [currentExperimentName, ...obj.hierarchy];
+      if(obj.hierarchy === undefined){
+        obj.hierarchy =[currentExperimentName]
+      } else {
+        obj.hierarchy = obj.hierarchy.includes(currentExperimentName) ? obj.hierarchy : [currentExperimentName,...obj.hierarchy]
+      }
```
Comment (ddelpiano) [inline client/src/helpers/ImagesTreeViewHelpers.js:37]:
```
if (obj.experiment === undefined || obj.experiment === true) {
    // experiment is true
    if(obj.hierarchy === undefined){ // no hierarchy key
      obj.hierarchy = [currentExperimentName]
    } else {
      // hierarchy key exists
      obj.hierarchy = obj.hierarchy.includes(currentExperimentName) ? obj.hierarchy : [currentExperimentName,...obj.hierarchy]
    }
} else {
    // experiment is false
    obj.hierarchy = ['others']
}

return obj;
```

is this doing the same? if yes, could we do a clean up please? Thanks

**Review by ddelpiano — CHANGES_REQUESTED** [review]

File: client/src/helpers/ImagesTreeViewHelpers.js (hunk 2 above)
Comment (Salam-Dalloul) [reply client/src/helpers/ImagesTreeViewHelpers.js:21]: we have cases where we dont have the experiment key at all where it consider it as experiment = false, and in this case we have to consider it as true, that's why i have made the logic separated in the first place depends if the  experiment exists or not by checking obj.hierarchy === undefined

File: client/src/helpers/ImagesTreeViewHelpers.js (hunk 3 above)
Comment (Salam-Dalloul) [reply client/src/helpers/ImagesTreeViewHelpers.js:37]: @ddelpiano yes this works but I have added a small modification that would cover the case where we have the experiment is false but we have a hierarchy key exists, in the above one will consider it as  ['others'] but I have made the modification to take the original  hierarchy array without any changes

**Review by ddelpiano — APPROVED** [review]

**Pull request merged**